test/cpp/api/functional.cpp

```diff
@@ -554,3 +554,38 @@ TEST_F(FunctionalTest, SoftplusDefaultOptions) {
   ASSERT_EQ(y.sizes(), torch::IntArrayRef({size, size, size}));
   ASSERT_TRUE(torch::allclose(y, y_exp));
 }
+
+TEST_F(FunctionalTest, Softshrink) {
+  const auto size = 3;
+  for (const auto lambda : {0.0, 0.42, 1.0, 4.2, 42.42}) {
+    auto x = torch::linspace(-10.0, 10.0, size * size * size);
+    x.resize_({size, size, size}).set_requires_grad(true);
+    auto y = F::softshrink(x, SoftshrinkOptions(lambda));
```

Comment:
We can change this to
```suggestion
    auto y = F::softshrink(x, /*lambda=*/lambda);
```
to also test `SoftshrinkOptions`'s implicit constructor :D